[GTK][WPE] Use std::array to store GObject signal and property identifiers

aperezdc opened the pull request:
#### 9d3358a6d868c44ea21571cf773d5a4ab1bf4852
<pre>
[GTK][WPE] Use std::array to store GObject signal and property identifiers
<a href="https://bugs.webkit.org/show_bug.cgi?id=281667">https://bugs.webkit.org/show_bug.cgi?id=281667</a>

Reviewed by Carlos Garcia Campos.

Replace usage of raw arrays with std::array for GObject signal
and property identifiers. This allows removing a number of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} opt-outs, or in a few
files narrowing them down to a few lines of code only.

* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
(webkit_authentication_request_cancel):
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp:
(webkit_download_class_init):
(webkit_download_set_allow_overwrite):
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.cpp:
(webkit_editor_state_class_init):
(webkitEditorStateSetTypingAttributes):
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
(webkitFaviconDatabaseGetFaviconInternal):
(webkit_favicon_database_clear):
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
(webkit_input_method_context_class_init):
(webkit_input_method_context_set_input_purpose):
(webkit_input_method_context_set_input_hints):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_class_init):
(webkitNetworkSessionDownloadStarted):
* Source/WebKit/UIProcess/API/glib/WebKitNotification.cpp:
(webkit_notification_close):
(webkit_notification_clicked):
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.cpp:
(webkit_option_menu_close):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):
(webkit_settings_apply_from_key_file):
(webkit_settings_set_webrtc_udp_ports_range):
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.cpp:
(webkit_uri_scheme_response_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitWindowProperties.cpp:
(webkit_window_properties_class_init):
(webkit_window_properties_get_fullscreen):
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.cpp:
(webkit_color_chooser_request_class_init):
(webkit_color_chooser_request_set_rgba):
(webkit_color_chooser_request_finish):
(webkit_color_chooser_request_cancel):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkit_print_operation_class_init):
(webkit_print_operation_set_print_settings):
(webkit_print_operation_set_page_setup):
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.cpp:
(webkit_web_inspector_class_init):

Canonical link: <a href="https://commits.webkit.org/285400@main">https://commits.webkit.org/285400@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30c2fcbbf9e3e3b11f03e27cc912f184f07a0013

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72549 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51970 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25337 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/23758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59774 "Built successfully") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23580 "Failed to checkout and rebase branch from PR 35361") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/76733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/23758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75616 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/47037 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/62471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/43688 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22108 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/65553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/20296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78403 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16790 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/61/builds/23580 "Failed to checkout and rebase branch from PR 35361") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16838 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/62484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/64825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/13114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/6758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11134 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47768 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/2553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/48837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/50127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/48580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
```diff
@@ -75,9 +75,7 @@ struct _WebKitAuthenticationRequestPrivate {
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-static guint signals[LAST_SIGNAL] = { 0, };
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+static std::array<unsigned, LAST_SIGNAL> signals { };
```
I guess with std::array we don't need the `{ } `no?

aperezdc replied:
Good point, thanks. I'll update the patch accordingly.

aperezdc committed:
[GTK][WPE] Use std::array to store GObject signal and property identifiers
https://bugs.webkit.org/show_bug.cgi?id=281667

Reviewed by Carlos Garcia Campos.

Replace usage of raw arrays with std::array for GObject signal
and property identifiers. This allows removing a number of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} opt-outs, or in a few
files narrowing them down to a few lines of code only.

* Source/WebKit/UIProcess/API/glib/WebKitAuthenticationRequest.cpp:
(webkit_authentication_request_cancel):
* Source/WebKit/UIProcess/API/glib/WebKitBackForwardList.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitCookieManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp:
(webkit_download_class_init):
(webkit_download_set_allow_overwrite):
* Source/WebKit/UIProcess/API/glib/WebKitEditorState.cpp:
(webkit_editor_state_class_init):
(webkitEditorStateSetTypingAttributes):
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
(webkitFaviconDatabaseGetFaviconInternal):
(webkit_favicon_database_clear):
* Source/WebKit/UIProcess/API/glib/WebKitInputMethodContext.cpp:
(webkit_input_method_context_class_init):
(webkit_input_method_context_set_input_purpose):
(webkit_input_method_context_set_input_hints):
* Source/WebKit/UIProcess/API/glib/WebKitNetworkSession.cpp:
(webkit_network_session_class_init):
(webkitNetworkSessionDownloadStarted):
* Source/WebKit/UIProcess/API/glib/WebKitNotification.cpp:
(webkit_notification_close):
(webkit_notification_clicked):
* Source/WebKit/UIProcess/API/glib/WebKitOptionMenu.cpp:
(webkit_option_menu_close):
* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_class_init):
(webkit_settings_apply_from_key_file):
(webkit_settings_set_webrtc_udp_ports_range):
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeResponse.cpp:
(webkit_uri_scheme_response_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_class_init):
* Source/WebKit/UIProcess/API/glib/WebKitWindowProperties.cpp:
(webkit_window_properties_class_init):
(webkit_window_properties_get_fullscreen):
* Source/WebKit/UIProcess/API/gtk/WebKitColorChooserRequest.cpp:
(webkit_color_chooser_request_class_init):
(webkit_color_chooser_request_set_rgba):
(webkit_color_chooser_request_finish):
(webkit_color_chooser_request_cancel):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkit_print_operation_class_init):
(webkit_print_operation_set_print_settings):
(webkit_print_operation_set_page_setup):
* Source/WebKit/UIProcess/API/gtk/WebKitWebInspector.cpp:
(webkit_web_inspector_class_init):

Canonical link: https://commits.webkit.org/285400@main
